src/ProjectFileTools.MSBuild/Worksapce.cs

```diff
@@ -145,5 +148,25 @@ private void MarkReload(object sender, FileSystemEventArgs e)
         {
             _needsReload = true;
         }
+
+        public void Dispose()
+        {
+            var watchers = Interlocked.Exchange(ref _watchers, null);
```

Comment:
Since `_watchers` is being nulled out, you will want to handle the case of other functions being called post-disposal. You can do this by making this throw `ObjectDisposedException` post-disposal.

Also, will need to hook up a way to call the `Dispose` method when the text view this eventually backs is being closed.